engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheReadingJobActor.scala

```diff
@@ -0,0 +1,101 @@
+import cromwell.engine.workflow.lifecycle.execution.callcaching.CallCacheReadActor._
+import cromwell.engine.workflow.lifecycle.execution.callcaching.CallCacheReadingJobActor._
+import cromwell.engine.workflow.lifecycle.execution.callcaching.EngineJobHashingActor.{CacheHit, CacheMiss, HashError}
+
+/**
+  * Receives hashes from the CallCacheHashingJobActor and makes requests to the database to determine whether or not there might be a hit
+  * for this job.
+  * 
+  * First receives the initial hashes, and asks the database if there is at least one entry with the same aggregated initial hash.
+  * If not, it's a CacheMiss.
+  * If yes, ask the CallCacheHashingJobActor for the next batch of file hashes.
+  * Every time a new batch of hashes is received, check against the database if at least one entry matches all those hashes.
+  * Keep asking for new batches until either one returns no matching entry, in which case it's a CacheMiss, or until it receives
+  * the last batch along with the aggregated file hash.
+  * In the latter case, asks the database for the first entry matching both the initial and aggregated file hash (if any).
+  * Sends the response to its parent.
+  * In case of a CacheHit, stays alive in case using the hit fails and it needs to fetch the next one. Otherwise just dies.
+  */
+class CallCacheReadingJobActor(callCacheReadActor: ActorRef) extends LoggingFSM[CallCacheReadingJobActorState, CCRJAData] {
+  
+  startWith(WaitingForInitialHash, CCRJANoData)
+  
+  when(WaitingForInitialHash) {
+    case Event(InitialHashingResult(_, aggregatedBaseHash), CCRJANoData) =>
+      callCacheReadActor ! HasMatchingInitialHashLookup(aggregatedBaseHash)
+      goto(WaitingForHashCheck) using CCRJAWithData(sender(), aggregatedBaseHash, None, 1)
+  }
+  
+  when(WaitingForHashCheck) {
+    case Event(HasMatchingEntries, CCRJAWithData(hashingActor, _, _, _)) =>
+      hashingActor ! NextBatchOfFileHashesRequest
+      goto(WaitingForFileHashes)
+    case Event(NoMatchingEntries, _) =>
+      cacheMiss
+  }
+  
+  when(WaitingForFileHashes) {
+    case Event(PartialFileHashingResult(hashes), _) =>
+      callCacheReadActor ! HasMatchingInputFilesHashLookup(hashes)
+      goto(WaitingForHashCheck)
+    case Event(CompleteFileHashingResult(_, aggregatedFileHash), data: CCRJAWithData) =>
+      callCacheReadActor ! CacheLookupRequest(AggregatedCallHashes(data.initialHash, aggregatedFileHash), data.currentHitNumber)
+      goto(WaitingForCacheHitOrMiss) using data.withFileHash(aggregatedFileHash)
+    case Event(NoFileHashesResult, data: CCRJAWithData) =>
+      callCacheReadActor ! CacheLookupRequest(AggregatedCallHashes(data.initialHash, None), data.currentHitNumber)
+      goto(WaitingForCacheHitOrMiss)
+  }
+  
+  when(WaitingForCacheHitOrMiss) {
+    case Event(CacheLookupNextHit(hit), data: CCRJAWithData) =>
+      context.parent ! CacheHit(hit)
```

Comment:
Ok, this answers my question about the CCRJA communicating with its parent.  😄  Wouldn't hurt to drop a comment back in the CCHJA, this is all kind of a lot to take in.